extensions/target-specific/chromium/ax-tree/ax-tree.js

```diff
@@ -11,18 +11,25 @@
 var AXTree = undefined;
 Loader.OnLoad(function() {
     var wrapperMaps = new Map();
-    var managersPromise = null;
+    var managers = null;
 
     function clearCaches() {
         wrapperMaps = new Map();
-        managersPromise = null;
+        managers = null;
     }
 
     function getManagers() {
-        if (managersPromise == null) {
-            managersPromise = DbgObject.global(Chromium.BrowserProcessSyntheticModuleName, "g_ax_tree_id_map").F("Object").array("Pairs").f("second");
+        if (managers == null) {
+            return DbgObject.global(Chromium.BrowserProcessSyntheticModuleName, "instance", "base::NoDestructor<ui::AXTreeManagerMap>").F("Object").f("map_").f("_List").array("Elements").f("second")
+            .catch(() => {
+                return DbgObject.global(Chromium.BrowserProcessSyntheticModuleName, "g_ax_tree_id_map").F("Object").array("Pairs").f("second");
+            })
+            .then((resolvedManagers) => {
+                managers = resolvedManagers;
+                return managers;
+            });
         }
-        return managersPromise;
+        return managers;
```

Comment:
This is kind of weird: sometimes `getManagers` returns a promise, sometimes it doesn't.  I'm guessing you went down this path because adding the `.catch` broke the `PromisedDbgObject` sugar?  I think better options are to either explicitly create a PromisedDbgObject from the promise, or change the callers to expect a regular Promise (which will resolve to a DbgObject) instead of a PromisedDbgObject.

Reply:
You're right, this was kind of weird. It was basically a workaround to maintain the existing caching behavior while still handling possible exceptions from DbgObject.global.

The change I just pushed adds a Promise.all() method that eliminates the weird behavior and makes it much cleaner.